MonstieWash/Assets/Scripts/MonsterAi/MonsterController.cs

```diff
@@ -121,4 +126,27 @@ public void AttackAnimationComplete()
             m_interruptedAnimation = null;
         }
     }
+
+    /// <summary>
+    /// Executes the logic required when a scene is completed (the last task is finished). Event is received from MonsterBrain <-- TaskTracker
+    /// </summary>
+    /// <param name="activeScene"> The scene that was completed. </param>
+    private void ProcessSceneComplete(Scene activeScene)
+    {
+        Debug.Log("MonsterController ProcessSceneComplete");
+        // More required logic can go here
+        StartCoroutine(PlayCompletionSparkles());
+    }
+
+    IEnumerator PlayCompletionSparkles()
+    {
+        var randTime = 0f;
+        foreach (var effect in completionEffectList)
+        {
+            randTime = UnityEngine.Random.Range(0.15f, 0.45f);
```

Comment:
again might be nice to expose this range to the designers, but up to you on this one since I don't imagine it would be changed that much.